### ISSUE TITLE
Failed for url

### ISSUE THREAD

**Issue by bobloco** [iam4x/pokemongo-webspoof#302]
Hi !

Everything was working smoothly but now I've got a problem on my phone when I click on the 'launch pokemon go" nothing happens but i get this message on Xcode : 
Pokemon WebSpoof[318:23032] -canOpenURL: failed for URL: "b335b2fc-69dc-472c-9e88-e6c97f84091c-3://" - error: "(null)

The weird part is that it's working well on another phone so I tried to delete the application on mine and installing it again but nothing seems to work.
Any solution to my problem ?

Thanks

**Comment by zrherda** [comment]
Running into the same issue.

**Comment by feichengh** [comment]
same for me

**Comment by MikeGarz** [comment]
Possibly the new update broke something?

**Comment by bittuahlawat** [comment]
Here are the steps to fix the issue for now

- Download to the [iOSAppInfo](https://github.com/wujianguo/iOSAppsInfo) project

- Select the following 4 files and add them over the pokemon-webspoof xcode project:  LMApp.h, LMApp.m, LMAppController.h, LMAppController.m

- In the Pokemon-webspoof project, select the ViewController.m file and look for the following method:

`- (IBAction)launchPokemonGo:(id)sender {
    
    if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"b335b2fc-69dc-472c-9e88-e6c97f84091c-3://"]]) {
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"b335b2fc-69dc-472c-9e88-e6c97f84091c-3://"]];
    }
    
}`


and replace it with the following:

`- (IBAction)launchPokemonGo:(id)sender {
    [[LMAppController sharedInstance] openAppWithBundleIdentifier:@"com.nianticlabs.pokemongo"];
}
`

**Comment by sbitswc** [comment]
I met same problem, follow bittuahlawat@'s procedure and add #import "LMAppController.h" at top of ViewController.m file. Should fix the problem. Thanks

**Comment by gingerrob1986** [comment]
I've have downloaded the new bits, how do I add them to the web spoof project please? I'm new to all this and I'm struggling.

**Comment by gingerrob1986** [comment]
I can not see this bit of code 
(IBAction)launchPokemonGo:(id)sender {

if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"b335b2fc-69dc-472c-9e88-e6c97f84091c-3://"]]) {
    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"b335b2fc-69dc-472c-9e88-e6c97f84091c-3://"]];
}

**Comment by bittuahlawat** [comment]
@gingerrob1986 you can drag and drop them to the files list from the other project to the webspoof project.  The code you are looking for is in the file called "ViewController.m" somewhere around line 40-ish.

**Comment by gingerrob1986** [comment]
OK, I've copied all those downloaded files to the project, I can not find that line, I might have to give up, but thanks.

**Comment by gingerrob1986** [comment]
![screen shot 2016-07-31 at 10 06 33](https://cloud.githubusercontent.com/assets/20751397/17275626/8787c82e-5706-11e6-8dc6-7256bd3cce24.png)

**Comment by goodasgood** [comment]
hey - 
i got that problem that the "launch pokemon go" button disappears when i follow your steps .. :/

**Comment by laureenmaeva** [comment]
I've done all steps but still not working.. Here's what's written on Xcode:

Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_LMAppController", referenced from:
      objc-class-ref in ViewController.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

I'm a novice and I don't know what this means.. Can someone help ?

**Comment by michael430** [comment]
Followed @bittuahlawat method , Launch Pokemon Go button is working now with the new update. 

Somehow I need to re-do the Accessibility of System Preferences/ Security & Privacy for the WebSpoof. After that seems everything is working as before.

Thanks bittuahlawat :)

Btw @gingerrob1986 you are looking for the wrong file where I saw your posted picture. You should able to find the code in "ViewController.m" file.

**Comment by bobloco** [comment]
@bittuahlawat Thanks for the solution it works perfectly now ! Hop everyone will get it to work too!

**Comment by absolut-times** [comment]
Thanks @bittuahlawat, works fine for me!

**Comment by gingerrob1986** [comment]
@michael430 yeah, it isn't in any of the folders, oh well, thanks though.

**Comment by gingerrob1986** [comment]
OK, found those lines, still doesn't work.
Do we think web spoof will be released with all this in at any point?

**Comment by paxuh** [comment]
I just get an error message saying; "use of undeclared identifier LMAppController when trying to replace the code. Why is that?

**Comment by AidanMcArdle** [comment]
@paxuh I had the same issue. 

> I just get an error message saying; "use of undeclared identifier LMAppController when trying to replace the code. Why is that?

means you haven't imported the file correctly. For me it was as simple as not having the closing " on the import statement at the top of that file.

#import "LMAppController.h"

Should be with the rest of the import statements like so

![like so](http://i.imgur.com/3ytIMPS.png)

**Comment by paxuh** [comment]
@AidanMcArdle Thanks a lot. It solved the issue.

**Comment by iruy** [comment]
Hi, I get this error:
```
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_LMAppController", referenced from:
      objc-class-ref in ViewController.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1
```

I can I fix this?

**Comment by gingerrob1986** [comment]
What am I doing wrong?

![screen shot 2016-07-31 at 16 06 49](https://cloud.githubusercontent.com/assets/20751397/17277350/d8645d2a-5738-11e6-904a-168d142bbb99.png)

**Comment by iruy** [comment]
@gingerrob1986 you have to right click on the folder and select "Add files.." and add those 4 files that @bittuahlawat mentioned

**Comment by gingerrob1986** [comment]
Thanks @iruy this seems to be my last issue

![screen shot 2016-07-31 at 16 16 53](https://cloud.githubusercontent.com/assets/20751397/17277423/41deaa70-573a-11e6-8325-cad869b66f0a.png)

**Comment by kebabortega** [comment]
i have a problem that after i added the files, I re-opened the webspoof app in mac,it doesn't build anymore because those 4 files were missing.  it seems that the location where the files were added is just temporary....

**Comment by iruy** [comment]
@kebabortega just read the last few post in this issue...

**Comment by amamut** [comment]
@gingerrob1986 I am getting the same issue.

**Comment by joshuanoon** [comment]
Everything seems fine for me after following @bittuahlawat steps, but I get left with the Launch Pokemon Go button on the Simulator, a seemly useable web spoof app and a stationary avatar on my iPhone screen. Any suggestions?

**Comment by dualshote** [comment]
Thanks a lot @bittuahlawat :D

**Comment by MuratV** [comment]
I receive following error on desktop app webspoof. Any idea why?

<img width="366" alt="screen shot 2016-07-31 at 23 37 47" src="https://cloud.githubusercontent.com/assets/11273414/17279571/d1a9b27a-5777-11e6-8672-d3d681602e02.png">

**Comment by fwartner** [comment]
Same here. as @MuratV

**Comment by MuratV** [comment]
@fwartner I solved it. I actually just forgot to go to xcode -> debug -> simulate location -> pokemonLocation.

**Comment by nvoniatis** [comment]
so, everything starts and runs on my phone, but just giving a blank white screen instead of "Launch PokemonGo" screen. any help ?
<img width="1012" alt="screen shot 2016-08-01 at 01 37 55" src="https://cloud.githubusercontent.com/assets/20760686/17279859/c96038d0-5788-11e6-902c-a5b1d5aa8e03.png">

**Comment by bittuahlawat** [comment]
@nvoniatis you see the blue pin right after "#import "LMAppController.h" on the left hand side.  That is called a breakpoint.  You can disable that breakpoint (by clicking on it again) or remove it (by dragging it to the right side).  Run your code again and it will work.

**Comment by nvoniatis** [comment]
@bittuahlawat ahh nevermind, fixed it! runs smoothly right now, but if close Xcode, everything is reset (even bundle identifier), any ideas? :/

**Comment by bittuahlawat** [comment]
@nvoniatis This is a short term (per session solution).  For it to be permanent, these files and code changes needs to be part of the package contents.  If @iam4x thinks this is the right fix for long term, maybe he can make it part of the release download.

**Comment by nvoniatis** [comment]
@bittuahlawat ahhh ok ok, thanks for your time mate, great work !

**Comment by Truro** [comment]
Hi guy ! I'm following bittuahlawat method but it can't work . 
I've got this error 
<img width="1274" alt="capture d ecran 2016-08-01 a 01 11 56" src="https://cloud.githubusercontent.com/assets/20760872/17280005/1a510ab6-5785-11e6-98cc-38d6685ac8eb.png">

How to fix it please ? sorry i'm totally noob to this x)

**Comment by bittuahlawat** [comment]
@Truro replace this:

<img width="748" alt="screen shot 2016-07-31 at 5 04 24 pm" src="https://cloud.githubusercontent.com/assets/4862554/17280264/e5e65fd6-5740-11e6-89f7-847abdb9e4b9.png">

with this:

<img width="826" alt="screen shot 2016-07-31 at 5 03 32 pm" src="https://cloud.githubusercontent.com/assets/4862554/17280258/bfcadee4-5740-11e6-8666-7bd0bd55ada9.png">

**Comment by Truro** [comment]
Thanks you @bittuahlawat 
but it still not working i've got this error this time 
<img width="769" alt="capture d ecran 2016-08-01 a 02 38 33" src="https://cloud.githubusercontent.com/assets/20760872/17280483/1b8940ea-5791-11e6-9e51-78592edf319c.png">

**Comment by bittuahlawat** [comment]
@Truro at the top of the file, make sure you you write "#import "LMAppController.h" next to other import statements.

<img width="435" alt="screen shot 2016-07-31 at 5 59 06 pm" src="https://cloud.githubusercontent.com/assets/4862554/17280660/91222b62-5748-11e6-97d0-abbd90dd3d4a.png">

**Comment by Truro** [comment]
Yes it's ok thank you for you time and your help @bittuahlawat   😁

**Comment by quattrodonkey** [comment]
@bittuahlawat Thanks for the input! I have a silly question though, after making the changes, how do I save them? I tried using "save" but when I reopen, all changes would disappear.

**Comment by bittuahlawat** [comment]
@quattrodonkey I mentioned this earlier, this is a per session solution.  Long term solution would be to have this part of the download of the project.  There is one thing you can do now to "save it" so you don't have to keep repeating these steps every time you launch the app.

**These steps will not do anything unless you follow the steps I have given above and have the project working**

Here are the steps:

- Right click on the project name and click on "Show in Finder"

<img width="519" alt="screen shot 2016-07-31 at 6 24 41 pm" src="https://cloud.githubusercontent.com/assets/4862554/17280965/93541b30-574c-11e6-90ba-38187a515b75.png">


- Copy the full "xcode-project" folder that I have highlighted

<img width="770" alt="screen shot 2016-07-31 at 6 24 19 pm" src="https://cloud.githubusercontent.com/assets/4862554/17280986/c6c804a4-574c-11e6-8a89-b46cc295a502.png">

- Right click on the webspoof app, click on "Options" and then click on "Show in Finder"

<img width="340" alt="screen shot 2016-07-31 at 6 24 59 pm" src="https://cloud.githubusercontent.com/assets/4862554/17281005/fd7e6362-574c-11e6-919f-21ae7b457175.png">

- It will open a finder window with the webspoof app in it.  Right click on the app and click "Show Package Contents"

<img width="521" alt="screen shot 2016-07-31 at 6 25 18 pm" src="https://cloud.githubusercontent.com/assets/4862554/17281023/3b7c481e-574d-11e6-93b3-d3a25f3764a5.png">

- It will also open a new finder window.  Follow the following path shown in screenshot.  Replace the "xcode-project" folder with the folder you copied in step 2.

<img width="1261" alt="screen shot 2016-07-31 at 6 26 35 pm" src="https://cloud.githubusercontent.com/assets/4862554/17281034/65540186-574d-11e6-8e24-3f25b712ed64.png">


These steps will "save" these changes to your app and you won't have to keep redoing these steps for every session.  Now when you will relaunch the webspoof app, the files will be there every time.  Hopefully this helps.  Let me know if you run into any issues.

**Comment by quattrodonkey** [comment]
@bittuahlawat Thank you! It worked! Is it true that when I open the app, it only opens an instance of the Xcode project? So to make changes to the original content, I'll have to replace the original content with the instance that I've updated. Padrone me if it sounds silly, I'm new to Xcode. Thanks!

**Comment by bittuahlawat** [comment]
@quattrodonkey in a nutshell, yes.  The package contains an xcode project that opens up when you launch the app but any changes to that app can't be saved back.  In order to do so, you have to follow the app and replace the packaged files within the changed files from the current instance.  Glad to hear it worked for you.

**Comment by gingerrob1986** [comment]
So is there no answer to my issue with location? 
![screen shot 2016-07-31 at 16 16 53](https://cloud.githubusercontent.com/assets/20751397/17288497/51d4b2a6-57cd-11e6-9aab-b684e3b08dd2.png)

**Comment by iruy** [comment]
@gingerrob1986 did you set the privacy settings for webspoof? Allowed Profiles in your phone general settings? Do you get any error, or just a warning? If it is just a warning, it should run the same

**Comment by gingerrob1986** [comment]
I did the privacy things, but that message keeps appearing so I can't do anything.

**Comment by gingerrob1986** [comment]
The gpx location file is invalid, what can i do?

**Comment by Klatreguy** [comment]
Hi. 
I did all the steps, and its running, also managed to save the stuff for everytime i open spoof.
BUT know i get the errorcode -1719 all the time and my dude wont move. Any ideas what i did wrong?

**Comment by paxuh** [comment]
@bittuahlawat I followed your steps to update long term save, but it gives me the following error message now; LMappController.h file not found and also all other files are missing in Xcode

**Comment by paxuh** [comment]
@bittuahlawat @Klatreguy I did also follow those steps for long term save, but I am also getting a plenty of -1719 error codes on web spoof, and it won't move anywhere.

**Comment by paxuh** [comment]
![skjermbilde 2016-08-01 kl 13 59 36](https://cloud.githubusercontent.com/assets/20754041/17293454/aad529c6-57f0-11e6-984e-ca2b20204e38.png)

**Comment by AidanMcArdle** [comment]
@Klatreguy @paxuh ensure you have gone into your Mac's System Preferences > Security & Privacy > Privacy tab, and given the application access.

![system preferences](http://i.imgur.com/M05qMps.png)

**Comment by paxuh** [comment]
I tried to reinstall this webspoof, but whenever I try implement files or change text in Bundle Identifier on Xcode, it won't save these things. Whenever I close and return back to Xcode project, the files are gone and Bundle Identifier i set back. Why is that?

**Comment by bullseye1975** [comment]
I added the changes and it builds fine.  It runs great on my iPhone.

But the UI with the map and controller never shows up on my Mac.  What am I doing wrong?

**Comment by Ladditron** [comment]
How do I download the files that were mentioned?

**Comment by skilfoy** [comment]
I'm spoofing the location just fine following bittuahlawat's instructions, BUT I can't move my character...

**Comment by Simbe90** [comment]
Does pulling the latest xcode-project from the master build and replacing it inside the pokemongo-webspoof app fix this issue without doing the fix manually as bittuahlawat instructed?

**Comment by Simbe90** [comment]
To answer my own question: Yes, that does work, it's updated on the latest master commit but not on the latest release yet.

**Comment by darksao** [comment]
Hi guys,

I have "use of undeclared identifier LMAppController" as issue. I'm pretty sure that I have well imported my files.

I used " Add files to pokemon-webspoof..." to add my files. But still, when I open Viewcontroller.m, it cannot find LMAppController.h ...
I also tried to add it by myself at the top but it doesn't work.

![capture d ecran 2016-08-06 00 24 (49](https://cloud.githubusercontent.com/assets/20621954/17452095/5e642fa4-5b6c-11e6-971c-77c334ef70f4.png)

Help me plz
